cabal-install/src/Distribution/Client/Security/HTTP.hs

```diff
@@ -119,11 +119,16 @@ mkRangeHeader from to = HTTP.Header HTTP.HdrRange rangeHeader
     rangeHeader = "bytes=" ++ show from ++ "-" ++ show (to - 1)
 
 mkReqHeaders :: [HC.HttpRequestHeader] -> Maybe (Int, Int) -> [HTTP.Header]
-mkReqHeaders reqHeaders mRange = concat [
+mkReqHeaders reqHeaders mRange' = concat [
       tr [] reqHeaders
     , [mkRangeHeader fr to | Just (fr, to) <- [mRange]]
     ]
   where
+    -- guard against malformed range headers.
+    mRange = case mRange' of
+        Just (fr, to) | fr >= to -> Nothing
+        _ -> mRange'
```

Comment:
Isn't this wrong? If we're being asked to get an empty range, I'd imagine the caller's expectation (of `getRange`) would be to end up with a `BodyReader` that reads an empty `ByteString`. By leaving out the range header, they'd instead get the whole file.

(Also seems like something that might be better handled in hackage-security.)

Reply:
this is fine because we already handle the case where we request a range but the server doesn't understand it and gives back the whole thing (the response code is different between the two cases)